### PR TITLE
Use updatable page source when effectivePredicate is None due to dynamic filters

### PR DESCRIPTION
**Pull request by raunaqmorarka**
Addresses similar issue as https://github.com/prestosql/presto/commit/2fac2761904e169d97200912928284b0d56182bd

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/connector/EmptySplitPageSource.java [hunk 1]
```diff
@@ -36,7 +35,7 @@ public void deleteRows(Block rowIds)
     @Override
     public CompletableFuture<Collection<Slice>> finish()
     {
-        return completedFuture(ImmutableList.of());
+        return completedFuture(Collections.emptyList());
```
Comment (sopel39) [inline presto-spi/src/main/java/io/prestosql/spi/connector/EmptySplitPageSource.java]: undo. Guava is smart when doing `copyOf` if underlying list intance is also immutable list

Comment (raunaqmorarka) [reply presto-spi/src/main/java/io/prestosql/spi/connector/EmptySplitPageSource.java]: I couldn't use Guava here due to the move to presto-spi
Would something else be more appropriate ?

Comment (sopel39) [reply presto-spi/src/main/java/io/prestosql/spi/connector/EmptySplitPageSource.java]: Ah, your are right.

**Comment by electrum** [comment]
I think it’d be cleaner to make `FixedPageSource` implement `UpdatablePageSource`. No need for two nearly identical classes in the SPI.

**Comment by sopel39** [comment]
> I think it’d be cleaner to make FixedPageSource implement UpdatablePageSource. No need for two nearly identical classes in the SPI.

I don't think you can simply make FixedPageSource implement UpdatablePageSource. How this should behave when there are splits present? For `EmptySplitPageSource` it's obvious (noop)

**Comment by electrum** [comment]
Makes sense. How about naming it `EmptyPageSource`? The “split” part seems related to a specific usage and not an inherent functionality.

**Comment by raunaqmorarka** [comment]
> Makes sense. How about naming it `EmptyPageSource`? The “split” part seems related to a specific usage and not an inherent functionality.

Renamed `EmptySplitPageSource` to `EmptyPageSource `

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**